src_cpp/cppBridge/NerlNIF.cpp

```diff
@@ -86,93 +92,77 @@ static ERL_NIF_TERM cppBridgeControllerDeleteModel_nif(ErlNifEnv* env, int argc,
 // Get weights
 static ERL_NIF_TERM get_weights_nif(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[])
 {
-
-    //std::vector<double> weights_list, bias_list, combined_list;
-    double* weights_list, *bias_list;
+    std::vector<double> weights_list, bias_list, combined_list;
+    std::vector<std::shared_ptr<ANN::Weights>> vec_of_weights_ptr;
     MatrixXd* weights_mat;
     VectorXd* bias;
+    std::vector<double> weights_list_vec;
+    std::vector<double> bias_list_vec;
+    ERL_NIF_TERM ret_tuple;
     //int mid; // TODO add it
-    std::vector<std::shared_ptr<ANN::Weights>> vec_of_weights_ptrs;
-
-    std::cout << "1!\n" << std::endl;
 
     // Get the singleton instance
     cppBridgeController *s = s->GetInstance();
 
-    std::cout << "2!\n" << std::endl;
-
     // Get the model from the singleton
     std::shared_ptr<SANN::Model> modelPtr = s-> getModelPtr(0);//TODO implement mid
 
-    std::cout << "3!\n" << std::endl;
-
     // Get the weights_list ptr
-    vec_of_weights_ptrs = modelPtr->get_weights_of_model();
-
-    std::cout << "4!\n" << std::endl;
+    vec_of_weights_ptr = modelPtr->get_weights_of_model();
 
-    // Go over all the weights and biases
-    for (std::vector<std::shared_ptr<ANN::Weights>>::iterator it = vec_of_weights_ptrs.begin() ; it != vec_of_weights_ptrs.end() ; it++)
+    // Go over all the weights and biases 
+    for (std::vector<std::shared_ptr<ANN::Weights>>::iterator it = vec_of_weights_ptr.begin() ; it != vec_of_weights_ptr.end() ; it++)
     {
-        std::cout << "5!\n" << std::endl;
-        std::cout << *it << "\n" << std::endl;
 
         // Go over the weights matrixXD and put it to a c++ vector
         weights_mat = (*it)->get_weights_mat_ptr();
 
-        std::cout << "6!\n" << std::endl;
-
-        //std::cout << "Rows:\n" << weights_mat->rows() << std::endl;
-        //std::cout << "Cols:\n" << weights_mat->cols() << std::endl;
-        //std::cout << "Here is the weights_mat:\n" << *weights_mat << std::endl;
+        // Create the result vector from the result matrix TODO: Native in Eigen optionally
 
-        // Convert MatrixXD to double*
-        //Map<MatrixXd>(weights_list, weights_mat->rows(), weights_mat->cols()) = *weights_mat;
-
-        weights_list = weights_mat->data();
-
-        std::cout << "7!\n" << std::endl;
-
-        //std::cout << "Here is the weights_list:\n" << weights_list[0] << std::endl;
-        //std::cout << "Here is the weights_list:\n" << weights_list[1] << std::endl;
-
-        std::cout << "8!\n" << std::endl;
+        for (int r = 0; r < weights_mat->rows(); r++){
+            for (int c = 0; c < weights_mat->cols(); c++){
+                weights_list_vec.push_back(weights_mat->coeff(r,c));
```

Comment:
before for:
weights_list_vec.resize(rows*cols); 

weights_list_vec[r*rows+c] = weights_mat->coeff(r,c);